### PR TITLE
Use new ordered set to allow for controlling strictMode

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

See all details on https://github.com/flame-engine/flame/pull/979
This is an alternative proposal which makes strictMode to be decided by the user

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by luanpotter** [comment]
cc @spydon @st-pasha

File: packages/flame/lib/src/components/component_set.dart [hunk 1]
```diff
@@ -36,9 +36,10 @@ class ComponentSet extends QueryableOrderedSet<Component> {
   final Component parent;
 
   ComponentSet(
-    int Function(Component e1, Component e2)? compare,
-    this.parent,
-  ) : super(compare);
+    int Function(Component e1, Component e2)? comparator,
+    this.parent, {
+    bool strictMode = true,
```
Comment (spydon) [inline packages/flame/lib/src/components/component_set.dart:41]: Why is the default true here, and false in `createDefault`?

**Review by spydon — APPROVED** [review]

File: packages/flame/CHANGELOG.md [hunk 2]
```diff
@@ -6,8 +6,8 @@
  - Method `PositionComponent.toRect()` now works for flipped/rotated components.
  - Make the root bundle exposed via `Flame.bundle` actually configurable
  - Take in an optional `Camera` as a parameter to `FlameGame`
- - It's no longer necessary to call `ComponentSet.register()` before `.query()`
  - Make super.onLoad mandatory to avoid memory leaks
+ - QueryableOrderedSet strictMode is configurable so you don't need to call `register` before `query`
```
Comment (wolfenrain) [inline packages/flame/CHANGELOG.md:10]:
```suggestion
 - `QueryableOrderedSet`'s `strictMode` is configurable so it is no longer necessary to call `register` before `query`
```

File: packages/flame/lib/src/components/component.dart [hunk 3]
```diff
@@ -332,7 +332,6 @@ class Component with Loadable {
   ///
   /// You can return a specific sub-class of OrderedSet, like
```
Comment (wolfenrain) [inline packages/flame/lib/src/components/component.dart:333]:
```suggestion
  /// You can return a specific sub-class of `OrderedSet`, like
```

Unrelated but bothers me. (Not sure if we can use `[]` here as that does require an import for dartdoc to resolve)

**Review by wolfenrain — APPROVED** [review]
LGTM

File: packages/flame/lib/src/components/component_set.dart (hunk 1 above)
Comment (luanpotter) [reply packages/flame/lib/src/components/component_set.dart:41]: my bad, should be true everywhere

File: packages/flame/lib/src/components/component_set.dart [hunk 4]
```diff
@@ -209,11 +196,13 @@ class ComponentSet extends QueryableOrderedSet<Component> {
   /// You must provide the parent so that it can be handed to the children that
   /// will be added.
   static ComponentSet createDefault(
-    Component parent,
-  ) {
+    Component parent, {
+    bool strictMode = false,
```
Comment (luanpotter) [inline packages/flame/lib/src/components/component_set.dart:200]:
```suggestion
    bool strictMode = true,
```

**Pull request merged**